src/controls/rotate.ts

```diff
@@ -0,0 +1,84 @@
+import { ControlCursorCallback, TransformActionHandler } from '../typedefs';
+import { radiansToDegrees } from '../util/misc/radiansDegreesConversion';
+import { NOT_ALLOWED_CURSOR } from './util';
+import { wrapWithFireEvent } from './wrapWithFireEvent';
+import { wrapWithFixedAnchor } from './wrapWithFixedAnchor';
+
+/**
+ * Find the correct style for the control that is used for rotation.
+ * this function is very simple and it just take care of not-allowed or standard cursor
+ * @param {Event} eventData the javascript event that is causing the scale
+ * @param {fabric.Control} control the control that is interested in the action
+ * @param {fabric.Object} fabricObject the fabric object that is interested in the action
+ * @return {String} a valid css string for the cursor
+ */
+export const rotationStyleHandler: ControlCursorCallback = (
+  eventData,
+  control,
+  fabricObject
+) => {
+  if (fabricObject.lockRotation) {
+    return NOT_ALLOWED_CURSOR;
+  }
+  return control.cursorStyle;
+};
+
+/**
+ * Action handler for rotation and snapping, without anchor point.
+ * Needs to be wrapped with `wrapWithFixedAnchor` to be effective
+ * @param {Event} eventData javascript event that is doing the transform
+ * @param {Object} transform javascript object containing a series of information around the current transform
+ * @param {number} x current mouse x position, canvas normalized
+ * @param {number} y current mouse y position, canvas normalized
+ * @return {Boolean} true if some change happened
+ * @private
+ */
+const rotateObjectWithSnapping: TransformActionHandler = (
+  eventData,
+  { target, ex, ey, theta, originX, originY },
+  x,
+  y
+) => {
+  const pivotPoint = target.translateToOriginPoint(
+    target.getRelativeCenterPoint(),
+    originX,
+    originY
+  );
+
+  if (target.lockRotation) {
+    return false;
+  }
+
+  const lastAngle = Math.atan2(ey - pivotPoint.y, ex - pivotPoint.x),
+    curAngle = Math.atan2(y - pivotPoint.y, x - pivotPoint.x);
+  let angle = radiansToDegrees(curAngle - lastAngle + theta);
+
+  if (target.snapAngle > 0) {
```

Comment:
Blame points to #3383

https://github.com/fabricjs/fabric.js/commit/10703055c0895b2bf0745c07da6024308322fda7#commitcomment-88268859

Reply:
Looking at #3383 it seems it was intended to be on FabricObject and that makes sense.
If so, should we add these values to json import/export?

Reply:
in json we don't import export behaviour data.
snapAngle was supposed to be on object, i remember that ( i think was contributed by stefan a friend of mine ), it was on object because we used to do fabric.Object.prototype.snapAngle = and set it to al objects.
Is also true that having it per object offers more flexibility ( imagine you want to rotate rects only by 90 degree, or you can have weird applications i don't want to dictate what are you doing with snapping )
So yes we could add it to the Interactivity mixin list of properties

Reply:
Maybe you should do that in #8401 so the conflict does rise?

Reply:
08a4bab27